### PR TITLE
Add filtering by jobName, projectName & userName to revisionJobsInfo endpoint

### PR DESCRIPTION
**Pull request by amouhoub**
- Add filtering by jobName (startWith pattern), projectName (startWith pattern) and userName (exact match pattern)
- Update tests

### CODE REVIEW

**Comment by fviale** [comment]
Before going further, I have the feeling that your branch is not uptodate with the master

**Comment by fviale** [comment]
Retest this please

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java [hunk 1]
```diff
@@ -392,11 +401,26 @@ private int getTotalNumberOfJobs(final DBJobDataParameters params) {
                     queryString.append("and parentId = null ");
                 }
 
+                if (!params.getJobName().isEmpty()) {
+                    queryString.append("and lower(jobName) like :jobName ");
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:405]: if you use lower on the database, you should also apply toLowerCase() to the parameter

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java [hunk 2]
```diff
@@ -392,11 +401,26 @@ private int getTotalNumberOfJobs(final DBJobDataParameters params) {
...
+                }
+
+                if (!params.getProjectName().isEmpty()) {
+                    queryString.append("and lower(projectName) like :projectName ");
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:409]: same remark

**Review by fviale — APPROVED** [review]
An important remark about lowercase

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java (hunk 1 above)
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:405]: Lower will break the index. This means that the search will have to go through all DB lines, are you sure we want this ?

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java [hunk 3]
```diff
@@ -233,12 +236,18 @@ public SchedulerDBManager(Configuration configuration, boolean drop) {
...
             }
-            if (user != null) {
+            if (user != null && !user.isEmpty()) {
                 criteria.add(Restrictions.eq("owner", user));
             }
             if (!childJobs) {
                 criteria.add(Restrictions.isNull("parentId"));
             }
+            if (jobName != null && !jobName.isEmpty()) {
+                criteria.add(Restrictions.like("jobName", jobName, MatchMode.START).ignoreCase());
```
Comment (mboussaa) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:246]: not sure if ignoreCase is a good idea

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java [hunk 4]
```diff
@@ -233,12 +236,18 @@ public SchedulerDBManager(Configuration configuration, boolean drop) {
...
+            }
+            if (projectName != null && !projectName.isEmpty()) {
+                criteria.add(Restrictions.like("projectName", projectName, MatchMode.START).ignoreCase());
```
Comment (mboussaa) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:249]: same

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java (hunk 1 above)
Comment (mboussaa) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:405]: I suggest also to not be case sensitive

Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:405]: @mboussaa you mean to be case sensitive
Case insensitive search has very bad performances.

It is possible to implement it (by duplicating a column inside the database with lowercase data, but this requires additional work, notably regarding backward compatibility)

**Review by mboussaa — APPROVED** [review]

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java (hunk 3 above)
Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:246]: indeed, there is still this ignore case after your changes

**Comment by amouhoub** [comment]
retest this please

**Comment by amouhoub** [comment]
@fviale @mboussaa  I have retriggered Jenkins manually and it has succeeded this time :
http://jenkins.activeeon.com/job/scheduling-pull-requests/3843/
However, as usual, GitHub isn't sync with Jenkins....

**Comment by fviale** [comment]
@amouhoub let's wait for the build to succeed again.

Once done you can merge it. Don't forget to squash your commits.

**Comment by amouhoub** [comment]
Retest this please

**Comment by amouhoub** [comment]
Retest this please

**Comment by amouhoub** [comment]
Retest this please

**Comment by fviale** [comment]
Retest this please

**Comment by fviale** [comment]
add to whitelist

**Pull request merged**